### PR TITLE
docs: add identity providers (admin) & remove identity providers (install)

### PR DESCRIPTION
**Pull request by stefanicjuraj**
Fixes https://github.com/daytonaio/enterprise-docs/issues/123
Fixes https://github.com/daytonaio/enterprise-docs/issues/177
Fixes https://github.com/daytonaio/enterprise-docs/issues/178

### CODE REVIEW

**Review by ivan-burazin — APPROVED** [review]

**Review by fabjanvucina — CHANGES_REQUESTED** [review]
@ivan-burazin We no longer need instructions on how to use Keycloak for managing identity providers. The admin dashboard in Daytona now fully supports management of these providers.

@stefanicjuraj You should transfer over instructions on how to create oauth applications for each IDP from the /installation/identity-providers page. After that, you can remove that page from the docs.

**Comment by fabjanvucina** [comment]
@stefanicjuraj Also, the redirect URI field has been added to the IDP form.

image

**Review by ivan-burazin — CHANGES_REQUESTED** [review]
@stefanicjuraj what is the diff with INSTALLATION IDPs and ADMIN IDPs

**Comment by stefanicjuraj** [comment]
> @stefanicjuraj what is the diff with INSTALLATION IDPs and ADMIN IDPs

There will only be administration identity providers for admins to be able to enable certain identity providers, and the users can refer to using them by navigating to the git providers section. I think that's the main gist provided by Fabijan's thoughts:

> @stefanicjuraj You should transfer over instructions on how to create oauth applications for each IDP from the /installation/identity-providers page. After that, you can remove that page from the docs.

**Comment by ivan-burazin** [comment]
Yes so we should delete the section under INSTALLATION then?

File: src/content/docs/index.mdx [hunk 1]
```diff
@@ -44,9 +44,9 @@ import DocumentListItem from "@components/DocumentListItem.astro";
     href="about/getting-started"
   />
   <DocumentListItem
```
Comment (fabjanvucina) [inline src/content/docs/index.mdx:46]: Remove this list item

File: src/content/docs/administration/identity-providers.mdx [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+- **Client Secret**
+
+    Enter the client secret registered with the Identity Provider.
+
+- **Account Linking Only**
+
+    Toggle to enable the Identity Provider for logging in to existing accounts only.
+
+4. Click **`Save`** to add **`GitHub`** as the Identity Provider.
+
```
Comment (fabjanvucina) [inline src/content/docs/administration/identity-providers.mdx:166]: @stefanicjuraj For each IdP, we need the instructions on how to create the OAuth app.

These instructions were already written in `src/content/docs/installation/idps.mdx`, they just need to ported into this new page.

Example (github):

3. **Configure GitHub as an Identity Provider**
- Under the default realm, click on **`Identity Providers`** under the *Configure* section on the left sidebar.
- Then click on **`github`**. On the next page, copy the **`Redirect URI`** and note it down, as we will need it later. *Don't close this tab*.

4. **Go to Developer Settings**
- Open [Github](https://github.com) in your browser, on the top right side of this page, click on your profile picture. Under the dropdown window, select **`Settings`** (at the bottom). Select **`Developer setting`s** on the left sidebar.

5. **Create OAuth Apps**
- Select the **`OAuth Apps`** on the left sidebar. Then click on **`New OAuth App`** on the top right side.

6. **Required Inputs for GitHub OAuth**

- For the application name, you can name it as easily recognizable for yourself and distinguish it from your other applications (recommend using deployed domain).
- The **`Homepage URL`** would be the URL to your Daytona dashboard.
- The **`Authorization callback URL`** is the URL you have copied from step 2, `Redirect URI`. You don't need to toggle `Enable Device Flow` here. Then click on **`Register application`**.

7. **Copy Client ID and Secrets**
- Under your application, click on the **`Generate a new client secret`** in the *Client secrets* section.
- Copy the **`credentials under Client ID`** and paste it into your Keycloak *Provider details* page in the Client ID section.
- Go back to the Github OAuth Application page, copy the **`credentials under the Client secrets`** section, and then paste it into *Client Secret* on your Keycloak *Provider details* page.

Of course, these instructions have to be reworded wherever Keycloak is mentioned.

**Review by fabjanvucina — CHANGES_REQUESTED** [review]

File: src/content/docs/administration/identity-providers.mdx (hunk 2 above)
Comment (stefanicjuraj) [reply src/content/docs/administration/identity-providers.mdx:220]: The old version of installation/idps section now has some outdated or missing identity providers, such as Azure Active Directory (Microsoft Entra), Azure DevOps, and Microsoft.

I still need to update the docs to reflect the new configuration of identity providers.

**Comment by ivan-burazin** [comment]
@stefanicjuraj ping me when review needed

**Review by ivan-burazin — APPROVED** [review]

File: src/content/docs/administration/identity-providers.mdx [hunk 3]
```diff
@@ -0,0 +1,543 @@
...
+    The client secret registered with the Identity Provider.
+
+- **Account Linking Only**
+
+    Toggle to enable the Identity Provider for logging in to existing accounts only.
+
+4. Click **`Save`** to add **`GitHub`** as the Identity Provider.
+
+Upon adding GitHub as an Identity Provider, you can configure GitHub OAuth to enable users to authenticate with Daytona using their GitHub account.
```
Comment (fabjanvucina) [inline src/content/docs/administration/identity-providers.mdx:244]: Before adding GitHub as an Identity Provider, you need to register an OAuth client to enable users to authenticate to Daytona using their GitHub account.

_make this change in all of the providers_

File: src/content/docs/administration/identity-providers.mdx [hunk 4]
```diff
@@ -0,0 +1,543 @@
...
+
+### GitHub OAuth
```
Comment (fabjanvucina) [inline src/content/docs/administration/identity-providers.mdx:246]: GitHub OAuth Client

File: src/content/docs/administration/identity-providers.mdx [hunk 5]
```diff
@@ -0,0 +1,543 @@
...
+
+1. Open [GitHub](https://github.com) and navigate to **`Settings`** > **`Developer settings`**.
+2. Select the **`OAuth Apps`** tab from the sidebar, and click **`New OAuth App`**.
+3. Enter the details for the OAuth App.
+
+    - `Application name`
+
+    - `Homepage URL` must be set to the URL of your Daytona installation.
+
+    - `Authorization callback URL` must be set to the `Redirect URI` used when connecting GitHub as an Identity Provider.
```
Comment (fabjanvucina) [inline src/content/docs/administration/identity-providers.mdx:258]: must be set to the `Redirect URI` value found in the form when adding GitHub as an Identity Provider.

File: src/content/docs/administration/identity-providers.mdx [hunk 6]
```diff
@@ -0,0 +1,543 @@
...
+
+4. Click **`Register application`** to create the GitHub OAuth App.
+
```
Comment (fabjanvucina) [inline src/content/docs/administration/identity-providers.mdx:261]: We need an another item here that says that the user should copy over the client id and the client secret values to finalize adding GitHub as an IdP.

Please make sure that these "GitHub" changes are reflected accordingly in other IdPs as well

File: src/content/docs/administration/identity-providers.mdx [hunk 7]
```diff
@@ -0,0 +1,543 @@
...
+4. Click **`Create`** to create the organization.
+5. Enter the details for the new project.
+
+    - `Project name` must be unique.
+
+    - `Visibility` must be set to `Private` or `Public`.
+
+6. Click **`Create`** to create the project.
+
```
Comment (fabjanvucina) [inline src/content/docs/administration/identity-providers.mdx:106]: There needs to be info where to find client_id, client_secret and tenant_id and that they need to be copied over to the form for adding the IdP

Comment (fabjanvucina) [reply src/content/docs/administration/identity-providers.mdx:106]: This logic applies for other IdP-s as well. Also, keep in mind that some IdP-s have specific config properties (for example tenant_id for Azure)

**Review by fabjanvucina — CHANGES_REQUESTED** [review]

**Comment by ivan-burazin** [comment]
@fabjanvucina are we good with this?

**Pull request merged**